Common/include/toolboxes/graph_toolbox.hpp

```diff
@@ -0,0 +1,489 @@
+/*!
+ * \file graph_toolbox.hpp
+ * \brief Functions and classes to build/represent sparse graphs or sparse patterns.
+ * \author P. Gomes
+ * \version 7.0.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "C2DContainer.hpp"
+
+#include <set>
+#include <vector>
+#include <limits>
+#include <cassert>
+#include <algorithm>
+
+/*!
+ * \enum ConnectivityType
+ * \brief In FVM points are connected by the edges (faces) of the grid.
+ *        In FEM, two points are connected if they have an element in common.
+ */
+enum class ConnectivityType {FiniteVolume=0, FiniteElement=1};
+
+
+/*!
+ * \class CCompressedSparsePattern
+ * \brief A simple class to store adjacency information in a
+ * compressed format suitable for sparse matrix operations.
+ * If built for row-major storage the inner indices are column indices
+ * and the pattern should be used as (row,icol), otherwise as (col,irow).
+ */
+template<typename Index_t>
+class CCompressedSparsePattern {
+  static_assert(std::is_integral<Index_t>::value,"");
+
+private:
+  su2vector<Index_t> m_outerPtr; /*!< \brief Start positions of the inner indices for each outer index. */
+  su2vector<Index_t> m_innerIdx; /*!< \brief Inner indices of the non zero entries. */
+  su2vector<Index_t> m_diagPtr;  /*!< \brief Position of the diagonal entry. */
```

Comment:
I created this light class to old generic sparse patterns, it is mostly an aggregate of containers.